### PR TITLE
Fix invalid handling of List<String> in spring-data-jpa custom queries

### LINKED ISSUE #13044: Spring Data JPA error when selecting property value
**Describe the bug**
I'm migrating an application which uses Spring Data Repostiories from Wildfly to Quarkus and encounter exceptions during the initialization phase.

The following code:

    @Query("SELECT DISTINCT code.otpsubject.id FROM OtpCodeEntity code")
    List<String> findAllOtpsubjectIds();
Causes: IllegalStateException: java.lang.String was not part of the Quarkus index

But `OtpSubject.id` is of type `String`. 
I don't exaclty understand  why I get that error. It doesn't even work if I remove `.id` from the query and change the return type to `List<OtpCodeEntity`.
That query could probably written differently but with more complicated queries that won't work.

**Expected behavior**
Same behavior as the non-quarkus spring-data-jpa (the example above works without exception).
Support for more complex queries (see: https://docs.jboss.org/hibernate/stable/entitymanager/reference/en/html/queryhql.html#queryhql-select)
Support of JPA queries like in https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.at-query
Support for all return types in https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-return-types

**Actual behavior**
`IllegalStateException: java.lang.String was not part of the Quarkus index` is thrown during initialization.

**To Reproduce**

https://github.com/renegrob/spring-data-api-issues

1. Execute `./mvnw compile quarkus:dev` on the command line

Error in the console output:
```
java.lang.RuntimeException: 
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.spring.data.deployment.SpringDataJPAProcessor#build threw an exception: java.lang.IllegalStateException: java.lang.String was not part of the Quarkus index
```

**Screenshots**
n/a

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
$ uname -a
Linux automatix 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
$ java -version
openjdk version "11.0.9" 2020-10-20 LTS
OpenJDK Runtime Environment Zulu11.43+21-CA (build 11.0.9+11-LTS)
OpenJDK 64-Bit Server VM Zulu11.43+21-CA (build 11.0.9+11-LTS, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
`1.9.1.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
./mvnw --version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/gro/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.9, vendor: Azul Systems, Inc., runtime: /usr/lib/jvm/zulu11-ca-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-52-generic", arch: "amd64", family: "unix"
```

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #13044

### CODE REVIEW

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java [hunk 1]
```diff
@@ -383,7 +384,7 @@ private Type verifyQueryResultType(Type t) {
                 }
                 return t;
             }
-        } else if (!DotNames.OBJECT.equals(t.name())) {
+        } else if (!DotNames.OBJECT.equals(t.name()) && !DotNames.STRING.equals(t.name()) && !isIntLongOrBoolean(t.name())) {
```
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java:387]: Minor gripe: if I understand the logic of this method right, I think you already handle the `isIntLongOrBoolean` case at the beginning of the method. And I think we should better handle all these special cases there.

**Review by gsmet — COMMENTED** [review]
I added a small comment. Feel free to ignore!

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java:387]: You are absolutely correct! I'll fix it

**Review by gsmet — APPROVED** [review]
Perfect, thanks!

**Comment by geoand** [comment]
:+1:

**Pull request merged**